restorer.go

```diff
@@ -87,3 +114,85 @@ func (r *Restorer) restoreLayer(cache Cache, sha string) error {
 
 	return layers.Extract(rc, "")
 }
+
+func (r *Restorer) analyzeLayers(appMeta LayersMetadata, meta CacheMetadata) error {
```

Comment:
It seems this is duplicating a fair amount of logic in analyzer.go. Is there a way to extract it into a function that could be shared by both phases? (That said, maybe there's value in keeping the code separate, so that it could more easily diverge?)

Reply:
This logic actually _moved_ from analyzer to restorer. Analyzer is currently using this same functions